test/manual/disk_space_monitor_test.cc

```diff
@@ -0,0 +1,196 @@
+    } catch (...) {
+        ex = std::current_exception();
+    }
+    co_await os.close();
+    if (ex) {
+        co_await coroutine::return_exception_ptr(std::move(ex));
+    }
+}
+
+static future<sstring> read_file_contiguous(const fs::path& path) {
+    auto f = co_await open_file_dma(path.native(), open_flags::ro);
+    sstring res = co_await with_closeable(make_file_input_stream(f), [] (input_stream<char>& in) {
+        return util::read_entire_stream_contiguous(in);
+    });
+    // Strip trailing white space
+    while (!res.empty() && std::isspace(res[res.size() - 1])) {
+        res.resize(res.size() - 1);
+    }
+    co_return res;
+}
+
+static future<> remove_file_nothrow(const fs::path& path) {
+    try {
+        co_await remove_file(path.native());
+    } catch (...) {
+        testlog.error("Could not remove {}: {}", path.native(), std::current_exception());
+    }
+}
+
+// return cmd standard output on success,
+// or throws an error containing the cmd standard error.
+static future<sstring> run_system_command(tmpdir& dir, sstring cmd) {
+    fs::path out_path = dir.path() / fmt::to_string(utils::make_random_uuid());
+    fs::path err_path = dir.path() / fmt::to_string(utils::make_random_uuid());
+    sstring err;
+    sstring out;
+    auto redirect_cmd = fmt::format("{} >{} 2>{}", cmd, out_path.native(), err_path.native());
+    auto status = std::system(redirect_cmd.c_str());
+    if (status) {
+        err = fmt::format("{} failed: {}", cmd, read_file_contiguous(err_path).get());
+    } else {
+        out = read_file_contiguous(out_path).get();
+        testlog.debug("{}: {}", cmd, out);
+    }
+    remove_file_nothrow(out_path).get();
+    remove_file_nothrow(err_path).get();
+    if (!err.empty()) {
+        throw std::runtime_error(err);
+    }
+    co_return out;
+}
+
+SEASTAR_THREAD_TEST_CASE(test_disk_space_monitor_metrics) {
+    using namespace std::chrono;
+
+    tmpdir dir;
+
+    // Prepare test filesystem
+    auto image_file = dir.path() / "image";
+    uint64_t image_size = 16 << 20; // Minimum of 4096 blocks
+    write_file(image_file, image_size).get();
+
+    // new mkfs.xfs does not support <300MB file systems except for its own unit tests.
+    // temp workaround, add same env vars as said tests to force create this small fs.
+    // See: https://lkml.kernel.org/linux-xfs/Yv2A9Ggkv%2FNBrTd4@magnolia/
+    size_t block_size = 1024;
+    sstring mkfs_cmd = fmt::format("TEST_DIR=1 TEST_DEV=1 QA_CHECK_FS=1 mkfs.xfs -b size={} -f '{}' -m crc=0,finobt=0", block_size, image_file.native());
+    run_system_command(dir, mkfs_cmd).discard_result().get();
+
+    // setup a new loop device
+    sstring losetup_setup_cmd = fmt::format("sudo losetup --find --show '{}'", image_file.native());
+    auto loop_dev = run_system_command(dir, losetup_setup_cmd).get();
+    auto detach_loop_dev = defer([&] () noexcept {
+        sstring losetup_detach_cmd = fmt::format("sudo losetup --detach '{}'", loop_dev);
```

Comment:
It's really bad practice to sudo without warning.

Reply:
What kind of warning do you have in mind?
Just a log message?

Reply:
No, using sudo on behalf of the user without the user knowing beforehand that this is what is going to happen.

Let's drop the test and understand that this is not practically testable.

Reply:
(at least in unit tests)

Reply:
okay, will drop unit test